### ISSUE TITLE
Support route parameters

### ISSUE THREAD

**Issue by joeldenning** [single-spa/single-spa-layout#9]
```js
import { constructRoutes } from 'single-spa-layout';

const routes = constructRoutes({
  routes: [
    { type: 'route', path: "/user/:id/settings", routes: [] },
  ]
})
```

**Comment by victorlevasseur** [comment]
The parameter's name is kind of useless except for the code's clarity ? It would be no use for single-spa-layout itself ? 

It asks the question about route priorities too : 

```js
const routes = constructRoutes({
  routes: [
    { type: 'route', path: "/user/new", routes: [] },
    { type: 'route', path: "/user/:id/settings", routes: [] },
  ]
})
```

When the user visits /user/new, I guess it should match the first one only. But theoretically, both route are matching.

**Comment by joeldenning** [comment]
Hi @victorlevasseur thanks for your thoughts.

> When the user visits /user/new, I guess it should match the first one only. But theoretically, both route are matching.

In the example code, this actually wouldn't be true since `/user/new` doesn't match `/user/:id/settings`. However, if you had both `/user/new` and `/user/:id` as routes, then yes you're totally right that we'd need to cover that.

> From what I read from the matchRoute.js code, a naive implementation to match parameters would make the two route matching. Solutions are :

Good ideas - I am hoping to use single-spa's [pathToActiveWhen](https://single-spa.js.org/docs/api#pathtoactivewhen) function for this, which was recently implemented in [5.3.0](https://github.com/single-spa/single-spa/releases/tag/v5.3.0) and exposed as a public single-spa API in https://github.com/single-spa/single-spa/releases/tag/v5.4.0

**Comment by joeldenning** [comment]
> The parameter's name is kind of useless except for the code's clarity ? It would be no use for single-spa-layout itself ?

Good question, perhaps we could add the route params as [single-spa props](https://single-spa.js.org/docs/api#registerapplication) to all applications within the route? That would make them not be useless anymore.
